tox.ini

```diff
@@ -11,10 +11,11 @@ deps =
     portend
     pytest-services
     munch
+passenv = PYTEST_ADDOPTS
+setenv =
+    PYTEST_ADDOPTS = --cov=twine --cov-context=test {env:PYTEST_ADDOPTS:--cov-report term-missing --cov-report html}
 commands =
-    pytest --cov=twine --cov-context=test \
-        --cov-report term-missing --cov-report html \
-        {posargs:tests}
+    pytest {posargs:tests}
```

Comment:
This feels somewhat complex, but addresses a small pain point of noisy coverage reports when running a subset of tests. An alternative might be to have a separate `coverage` testenv, e.g.

https://github.com/sphinx-doc/sphinx/blob/master/tox.ini
https://github.com/jazzband/pip-tools/blob/master/tox.ini

Reply:
Yes, I think that showing the entire coverage report every time a single test is run is noisy, and I agree to having a `coverage` test env to get the coverage report via `tox -e coverage`, along with this change is a good way to separate those two out.

Reply:
The way this works, if someone overrides `PYTEST_ADDOPTS` they won't see coverage reports, no? Also now there are a few ways to pass things into pytest and that seems very confusing. Why don't we do:

```
commands =
    pytest {posargs:--cov=twine --cov-context=test --cov-report term-missing --cov-report html tests}
```

Reply:
> The way this works, if someone overrides `PYTEST_ADDOPTS` they won't see coverage reports, no?

Yes, that was part of the point: allowing `testenv` to be flexible enough to run the whole test suite with coverage reports, and also a single test with easily visible results. For example, if someone wishes to generate only HTML reports, they could set:

```
export PYTEST_ADDOPTS='-k "not integration" --cov-report html'
```

However:

> Why don't we do:
> 
> ```
> commands =
>     pytest {posargs:--cov=twine --cov-context=test --cov-report term-missing --cov-report html tests}
> ```

Yep, I like that. Done in https://github.com/pypa/twine/pull/633/commits/a4a014239f55150ef4b831835f7ff66ffb86be57.

I left the `PYTEST_ADDOPTS` bit in for skipping integration tests. It also allows folks to tweak their reporting settings, e.g.

```
export PYTEST_ADDOPTS='-k "not integration" --tb=line --show-capture=no'
```